src/coreclr/gc/unix/gcenv.unix.cpp

```diff
@@ -918,18 +917,12 @@ static size_t GetLogicalProcessorCacheSizeFromOS()
 
             if (ReadMemoryValueFromFile(path_to_size_file, &cache_size_from_sys_file))
             {
-                // uint64_t to long conversion as ReadMemoryValueFromFile takes a uint64_t* as an argument for the val argument.
-                size = (long)cache_size_from_sys_file;
-                path_to_level_file[index] = (char)(48 + i);
+                cacheSize = std::max(cacheSize, (size_t)cache_size_from_sys_file);
```

Comment:
doesn't seem worth it to me to depend on theoretical case not happening when you can so trivially handle it and make it consistent from what it's done so far in this function.

Reply:
> make it consistent from what it's done so far in this function.

It is done in one place in this function that needs it. It is not consistently done in this function.

For example, https://github.com/dotnet/runtime/blob/398adddc2bed0d0b52187f613f578194c58d9dc8/src/coreclr/gc/unix/gcenv.unix.cpp#L977 does not protect against bogus values either.

If you would like to protect against bogus cache sizes, it would be best to do it at the very end of the function or even in the calling code in the GC. For example, ignore the cache size if it is more than 1GB.

Reply:
The point is that theoretically, that function could return any value if the kernel generated wrong contents of those virtual files due to some bug. So it doesn't seem to be useful to handle one specific case explicitly. If there was a bug, it can as well contain -2 or 1234567890 or whatever random number. As I've said, it is very different from the `sysconf` where the `-1` is valid and documented return value that we've just forgotten about.